### ISSUE TITLE
Ignore unused-binding metadata on one of the arity in multi-arity function is not respected

### ISSUE THREAD

**Issue by keychera** [clj-kondo/clj-kondo#2145]
**version**

2023.07.13

**macro usage**

no macros

**platform**

tested on native clj-kondo, MacOS

**editor**

also tested in Calvaa VSCode using 2023.06.02-SNAPSHOT

**problem**

ignore unused-binding metadata on one of the arity in multi-arity function is not respected

**repro**

file `test-kondo.clj`
```clojure
(defn foo
  ([p1] (foo p1 "p2"))
  #_{:clj-kondo/ignore [:unused-binding]}
  ([p1 p2] (println p1 "not using p2")))
```

```sh
clj-kondo --lint test-kondo.clj
# =>
# test-kondo.clj:4:8: warning: unused binding p2
# linting took 21ms, errors: 0, warnings: 1
```

**config**

not a config issue

**expected behavior**

linting will produce zero warning
